pkg/cli/cli.go

```diff
@@ -143,7 +141,7 @@ func NewDefaultOcCommand(o kubecmd.KubectlOptions) *cobra.Command {
 		if kcmdutil.CmdPluginAsSubcommand.IsEnabled() {
 			// Command exists(e.g. kubectl create), but it is not certain that
 			// subcommand also exists (e.g. kubectl create networkpolicy)
-			if _, ok := allowedCmdsSubcommandPlugin[foundCmd.Name()]; ok {
+			if kubecmd.IsSubcommandPluginAllowed(foundCmd.Name()) {
```

Comment:
+1 for bringing this in. 

Btw, would you mind looking at the plugin completion?

https://github.com/kubernetes/kubernetes/blob/1020678366f741834f06b9f0087e57e48f93ffba/staging/src/k8s.io/kubectl/pkg/cmd/cmd.go#L329

https://github.com/kubernetes/kubernetes/blob/1020678366f741834f06b9f0087e57e48f93ffba/staging/src/k8s.io/kubectl/pkg/cmd/cmd.go#L472

It seems we are missing it.

Reply:
Updated code with the shell completion for plugins. But this https://github.com/kubernetes/kubernetes/blob/1020678366f741834f06b9f0087e57e48f93ffba/staging/src/k8s.io/kubectl/pkg/cmd/cmd.go#L472 seems to be added in 1.29 and we can get this in in next bump

Reply:
Ah, I see. Thanks for adding the first part.